storage/storage-interface/src/finality_view.rs

```diff
@@ -0,0 +1,118 @@
+use std::sync::RwLock;
+
+use aptos_types::{ledger_info::LedgerInfoWithSignatures, transaction::Version};
+
+use crate::{AptosDbError, DbReader, Result};
+
+/// A wrapper over [`DbReader`], representing ledger at the end of a block
+/// as the latest ledger and its version as the latest transaction version.
+pub struct FinalityView<Db> {
+    reader: Db,
+    finalized_ledger_info: RwLock<Option<LedgerInfoWithSignatures>>,
+}
+
+impl<Db> FinalityView<Db> {
+    pub fn new(reader: Db) -> Self {
+        Self {
+            reader,
+            finalized_ledger_info: RwLock::new(None),
+        }
+    }
+}
+
+impl<Db: DbReader> FinalityView<Db> {
+    /// Updates the information on the latest finalized block's ledger.
+    pub fn set_finalized_ledger_info(&self, ledger_info: LedgerInfoWithSignatures) -> Result<()> {
```

Comment:
accept `blockheight` here instead of `ledger_info` then fetch the blockinfo from the DB reader and populate it inside this method.